### PR TITLE
Create parking_garage2x2.json

### PR DESCRIPTION
**Pull request by Maleclypse**
Update parking_garage4x4.json

#### Summary
None

#### Purpose of change
Adds more reasonable parking garages with ramps instead of elevators.

#### Describe the solution
Make parking garages bigger with ramps and no elevators.

#### Describe alternatives you've considered
Adding a people sized elevator may still do it. It won't be functional but it will add verisimilitude.

#### Testing
tbd

### CODE REVIEW

**Comment by LovamKicsiGazsii** [comment]
Wonderful change. The old parking garages were very impractical so I don't mind having them replaced with bigger, functional ones.

File: data/json/mapgen/parking_garage4x4.json [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+        "|                                           =***",
+        "|                                           =***",
+        "W                                           =***",
+        "W                                           =***",
+        "W                                           =***",
+        "W                                       ssss|***",
+        "|                                       s/si|***",
+        "||||||||||||====|====|====|====|====|====||||***"
+      ],
+      "palettes": [ "parking_palette" ],
```
Comment (Maleclypse) [inline data/json/mapgen/parking_garage4x4.json:353]:
```suggestion
      "palettes": [ "parking_garage4x4_palette" ],
```

File: data/json/mapgen/parking_garage4x4.json [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+        "|EEEEEEE!_______y____y____y____y____y____yss=***",
+        "WEEEEEEE!_______y____y____y____y____y____yss=***",
+        "WEEEEEEE!_______y____y____y____y____y____yss=***",
+        "WEEEEEEE!_______y____y____y____y____y____yss=***",
+        "WEEEEEEE!_______ssssssssssssssssssssssssssss=***",
+        "|EEEEEEE!_______ssssssssssssssssssssssssssss=***",
+        "|||||||||====================================***",
+        "************************************************"
+      ],
+      "palettes": [ "parking_palette" ],
```
Comment (Maleclypse) [inline data/json/mapgen/parking_garage4x4.json:400]:
```suggestion
      "palettes": [ "parking_garage4x4_palette" ],
```

File: data/json/mapgen/parking_garage4x4.json [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+        { "vehicle": "parking_garage", "x": 29, "y": 18, "chance": 25, "rotation": 90 },
+        { "vehicle": "parking_garage", "x": 34, "y": 18, "chance": 25, "rotation": 90 },
+        { "vehicle": "parking_garage", "x": 39, "y": 18, "chance": 25, "rotation": 90 }
+      ]
+    }
+  },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "om_terrain": [ [ "parking_garage4x4_2_0", "parking4x4_garage_2_1" ], [ "parking_garage4x4_2_2", "parking4x4_garage_2_3" ] ],
```
Comment (mqrause) [inline data/json/mapgen/parking_garage4x4.json:299]:
```suggestion
    "om_terrain": [ [ "parking_garage4x4_2_0", "parking_garage4x4_2_1" ], [ "parking_garage4x4_2_2", "parking_garage4x4_2_3" ] ],
```

**Comment by Maleclypse** [comment]
@LovamKicsiGazsii Do you have any recommendations for an easy way to place vehicles?  Going through and testing different combinations for every single parking slot is a lot of work.

**Comment by LovamKicsiGazsii** [comment]
I've started using `vehicles` as an alternative to `place_vehicles` when I have too many cars to position. I just have to assign a symbol for a vehicle groups' anchor point, place it somewhere on the map, and the game will do the coordinates for me.

It's the solution I used for the motels, as an example.

**Comment by Maleclypse** [comment]
@LovamKicsiGazsii  My ramps are all jacked up.  This works but it looks ugly as heck to me.  If you have some time could you review and help with this?  I think I need to change some of my car directions as well.  Or should I just get this in and hopefully someone will fix it later?  I'm also not sure it's spawning naturally.

**Comment by LovamKicsiGazsii** [comment]
> @LovamKicsiGazsii My ramps are all jacked up. This works but it looks ugly as heck to me. If you have some time could you review and help with this? I think I need to change some of my car directions as well. Or should I just get this in and hopefully someone will fix it later? I'm also not sure it's spawning naturally.

Yeah, sure. I just need to finish writing a PR, and I'll get back to this.

File: data/json/mapgen/parking_garage4x4.json [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+        "|              zz                   z       |***",
+        "|              z                    z       |***",
+        "|              z                    z       |***",
+        "|              z                    z       |***",
+        "|                       sssss       z       |***",
+        "|                       sIsIs       z       |***",
+        "|-----                  sssss               |***",
+        "|                                           |***",
+        "|                                    >>>>>>>|***",
+        "|                             sssss  OOOOOOO|***",
```
Comment (LovamKicsiGazsii) [inline data/json/mapgen/parking_garage4x4.json:228]:
```suggestion
        "|              z                    z||||||||***",
        "|              z                    zOOOOOOO|***",
        "|              z                    z>>>>>>>|***",
        "|                       sssss       z       |***",
        "|                       sIsIs       z       |***",
        "|-----                  sssss               |***",
        "|                                           |***",
        "|                                           |***",
        "|                             sssss         |***",
```

File: data/json/mapgen/parking_garage4x4.json [hunk 5]
```diff
@@ -0,0 +1,415 @@
...
+        "|                             sIsIs         |***",
+        "|                             sssss         |***",
+        "|-----          <o                          |***",
+        "|               <o                          |***",
+        "|               <o                          |***",
+        "|               <o                          |***",
+        "|               <o                          |***",
+        "W               <o                          |***",
+        "W               <o                          =***",
```
Comment (LovamKicsiGazsii) [inline data/json/mapgen/parking_garage4x4.json:237]:
```suggestion
        "|-----          <o|                         |***",
        "|               <o|                         |***",
        "|               <o|                         |***",
        "|               <o|                         |***",
        "|               <o|                         |***",
        "W               <o|                         |***",
        "W               <o|                         =***",
```

File: data/json/mapgen/parking_garage4x4.json [hunk 6]
```diff
@@ -0,0 +1,415 @@
...
+        "|              zz                   z       |***",
+        "|              z                    z       |***",
+        "|              z                    z       |***",
+        "|              z                    z       |***",
+        "|                       sssss       z       |***",
+        "|                       sIsIs       z       |***",
+        "|-----                  sssss               |***",
+        "|                                           |***",
+        "|                                    >>>>>>>|***",
+        "|                             sssss  OOOOOOO|***",
```
Comment (LovamKicsiGazsii) [inline data/json/mapgen/parking_garage4x4.json:299]:
```suggestion
        "|                                           |***",
        "|                             sssss         |***",
```
This is the topmost floor, no need for a ramp leading up here.

File: data/json/mapgen/parking_garage4x4.json [hunk 7]
```diff
@@ -0,0 +1,415 @@
...
+        "|                             sIsIs         |***",
+        "|                             sssss         |***",
+        "|                                           |***",
+        "|               <o                          |***",
+        "|               <o                          |***",
+        "|               <o                          |***",
+        "|               <o                          |***",
+        "|               <o                          |***",
+        "W               <o                          |***",
+        "W               <o                          =***",
```
Comment (LovamKicsiGazsii) [inline data/json/mapgen/parking_garage4x4.json:309]:
```suggestion
        "|               |O>                         |***",
        "|               |O>                         |***",
        "|               |O>                         |***",
        "|               |O>                         |***",
        "|               |O>                         |***",
        "|               |O>                         |***",
        "W               |O>                         |***",
        "W                                           =***",
```

**Review by FarkasKBence — COMMENTED** [review]
These issues are too complex for me to solve using only the web editor. It would be best if we get this PR merged, then I could quickly make a follow-up PR to fix the aforementioned issues.

Anyway, these ramps should be aligned now. Not the most elegant, but I could get my trusty city car to the top with these.

**Comment by LovamKicsiGazsii** [comment]
Also, could you replace the `4x4` with `2x2` in the IDs to avoid confusion?

**Comment by Maleclypse** [comment]
> Also, could you replace the `4x4` with `2x2` in the IDs to avoid confusion?

Yes, I realized that today I'll get that changed

**Pull request merged**